### PR TITLE
Tcga drugs

### PR DESCRIPTION
**Pull request by bwalsh**
This PR has one new feature:
* harvests gdc legacy endpoint for files containing treatments clinical data for tcga individuals
addresses #178 

```
$ ~/bin/swift list -l --prefix outputs/compound bmeg
      246587 2018-09-27 23:46:53         application/json outputs/compound/normalized.Compound.Vertex.json
    26452156 2018-09-27 23:46:53         application/json outputs/compound/normalized.HasEnvironment.Edge.json
    60841105 2018-09-27 23:46:53         application/json outputs/compound/normalized.ResponseTo.Edge.json
     2344243 2018-09-27 23:46:53         application/json outputs/compound/normalized.TreatedWith.Edge.json

$ ~/bin/swift list -l --prefix outputs/gdc bmeg
    68081920 2018-09-27 23:55:16         application/json outputs/gdc/Aliquot.Vertex.json
    65916062 2018-09-27 23:55:16         application/json outputs/gdc/AliquotFor.Edge.json
    18321199 2018-09-27 23:55:16         application/json outputs/gdc/Biosample.Vertex.json
    22931586 2018-09-27 23:55:16         application/json outputs/gdc/BiosampleFor.Edge.json
      211812 2018-09-27 23:55:16         application/json outputs/gdc/Compound.Vertex.json
     9711916 2018-09-27 23:55:16         application/json outputs/gdc/InProject.Edge.json
   105282110 2018-09-27 23:55:16         application/json outputs/gdc/Individual.Vertex.json
     2460520 2018-09-27 23:55:16         application/json outputs/gdc/TreatedWith.Edge.json
   371388416 2018-09-27 23:55:16 application/octet-stream outputs/gdc/requests_cache.sqlite

```

### CODE REVIEW

**Comment by bwalsh** [comment]
The following drug names were not found by the drug_enricher.

Name
--
06-BG (NABTT   0803)
06-BG
06BG
06GB
5 FU
5- FU
5-FU +   LEULOV
5-FU
5-FU+   etoposidium
5-Flourouracil
5-Fluoruoracil
5FU
5fluorouracil+oxaciplatina+l-folinian   discido
6 Thiguanine
81C6
81c6
9   immunoamino camptnetecin
9AC 9 Aminocamplotecian
A202171   Protocol
A4QN
ACT PEP3 KLH
AE 788
ATTAC
Abagovomag vs Placebo
Acctuane
Adrimycin
Almita
Amgen 706
Amitostin
Amitostine
Anti   necplatens
Armidex
B12
BA4 43 9006
BCG
BCNU
BS1-201
Bevacizamab/Placebo
Bevacozimab
Bevcizumab
CAI (NABTT   9712)
CAI (NABTT 97212)
CAI NABIT   9712
CCDP
CCNG
CCNU
CCNu
CDDP
CHOP
CPT11
CRA
CVAD
Capecetabine
Capecytabinum
Carbobplatin
Carbplatin
Catumaxumab
Cedicanib
Celbrex
Cepecitabine
Chemo,   Multi-Agent, NOS
Chemo, NOS
Cilenaitide
Ciplastin
Cisplatain
Copolang
Cyclophasphamide
Cyclophospamide
Cyotxan
Cytoxen
Decodron
Dexaethasone
Dexamethasome
Dexamethsone
Dexamthasone
Dexmethasone
Docetaxelum
Doxetaxel
Doxetaxol
Doxorubincin HCL
E-75
ETC Posidum
Epirubicoin
Erlotonib
FLOUROURACIL
FT
Fuluestrant
GEMICTIABINE
GOG 218
GOG182
Gaminocamptothecin
Gamzar
Gemar
Gemcitibane
Gemicitabine
Gemzan
Gemzet
Gezmar
Gliadle   Wafer
Gomzar
Hexamethymelamine
High dose ara-c
Hydoxyurea
Hydrochoroquine
Hydroxurea
Hydroxyuerea
Hydroyurea
Hyrdroxyurea
Hyroxyurea
ICE
ICT-107
IL 12
IL 13
IL 2
IL-13PE
IL-2 (high dose)
IL-2
INF
Ifostamide
Irintocean
Irintotecan
Irolfluvin
Ironotecan
Isotrectinoin (acccutane)
Karerltecin
Leuocvorin
Levcovorin
Levenracetam
Liosomal   Docorubicin
Liposomal Doxorubican
Lumustine
Lymphocyte Infusion
MAB I 131
MAB I-131
MAB I131
MABI131
MAGI131-81c6
MAb I-131
MEDT 575
MU81C6
Mayo 425-20
Melphalon
Motexatin   Gadoinium
Nexavaar
Not   otherwise specified
Not specified
O-ICE
O6B6
O6BG
OS1-774
PC2
PCB
PCV
PEP3 KLH
PLFE
PLFE/FLO
PTK-ZK
Pacilatxel
Paciltaxal
Paciltaxel
Paciltaxle
Pacitaxel
Pacliatxel
Pacliltaxel
Paclitaxil
Palixtaxel
Poly LCLC
Procarbizine
R04929097
RBBX 01
Rapamcyin
Ras Inhibitor
Reinoid/CIS   retinoic acid
Rhumab/Ind
SAHA
SCH 58500
SCH63666
SCH6636
Sargramostin
Sovatenib
Study drug   AMG 655
TC
Tamoxiten
Tanceva
Tarveca
Taxoterecin
Temazolomide
Temodor
Temoxolomide
Temozlomide
Temozolomode
Temozolomoide
Temozomide
Tomaxifen
Topecan
Toptecan
Trustuzumab
VP16
Vamydex
ac
adriamicin
anastrozolum
anti   neopastons
ch81c6
dcVax
letrozolum
metotreksat
mu81c6
paclitaxelum
rec MAGE 3-AS + AS15 ACS1 / Placebo Vaccine
rec   MAGE3-AS+AS15 ASCI vs Placebo
recMAGE3-AS+AS15 ASCI/Placebo vaccine
recPRAME+AS15   ASCI
recPRAME+AS15
tamoxiphen+anastrazolum
tamoxiphene
tamoxiphene+anastrozolum
tc

**Comment by kellrott** [comment]
What is the best way to start spot fixing this list?
For example `Acctuane` is probably `Accutane`

**Comment by bwalsh** [comment]
Perhaps use Google's  suggestion api we could leverage:

```
 curl -s http://suggestqueries.google.com/complete/search?client=firefox\&q=Flourouracil | jq .
[
  "Flourouracil",
  [
    "fluorouracil",
    "fluorouracil cream",
    "fluorouracil cream 5",
    "fluorouracil 5",
    "fluorouracil side effects",
    "fluorouracil 5 topical cream",
    "fluorouracil cost",
    "fluorouracil pictures",
    "fluorouracil package insert",
    "fluorouracil cream dosage"
  ]
]
```

**Comment by bwalsh** [comment]
@kellrott  Added source/drug_enricher/drug_alias.tsv  tcga drugs w/out ontologies went from 218 to 152

**Comment by bwalsh** [comment]
no ontology |
------- |
06-BG
06-BG (NABTT 0803)
06BG
06GB
5 FU
5- FU
5-FU
5-FU + LEULOV
5-FU+ etoposidium
5FU
5fluorouracil+oxaciplatina+l-folinian discido
81C6
81c6
9 immunoamino camptnetecin
9AC 9 Aminocamplotecian
A202171 Protocol
A4QN
ACT PEP3 KLH
AE 788
ATTAC
AZD
Abagovomag vs Placebo
Almita
Amitostin
Amitostine
Anti necplatens
B12
BA4 43 9006
BCG
BCNU
BS1-201
Bevacizamab/Placebo
Bevacozimab
CAI (NABTT 9712)
CAI (NABTT 97212)
CAI NABIT 9712
CCDP
CCNG
CCNU
CCNu
CDDP
CHOP
CRA
CVAD
Capecytabinum
Carbobplatin
Cedicanib
Chemo, Multi-Agent, NOS
Chemo, NOS
Cilenaitide
Copolang
Docetaxelum
Doxorubincin HCL
E-75
EMD
ETC Posidum
Epirubicoin
FT
GEMICTIABINE
GOG 218
GOG182
Gaminocamptothecin
Gemar
Gemzet
Gezmar
Gliadle Wafer
Gomzar
Hexamethymelamine
High dose ara-c
ICE
ICT-107
IL 12
IL 13
IL 2
IL-13PE
IL-2
IL-2 (high dose)
INF
Irintocean
Irolfluvin
Isotrectinoin (acccutane)
Karerltecin
Leuocvorin
Levenracetam
Liosomal Docorubicin
Lymphocyte Infusion
MAB I 131
MAB I-131
MAB I131
MABI131
MAGI131-81c6
MAb I-131
MEDT 575
MU81C6
Mayo 425-20
Motexatin Gadoinium
Not otherwise specified
Not specified
O-ICE
O6B6
O6BG
OS1-774
PC2
PCB
PCV
PEP3 KLH
PLFE
PLFE/FLO
PTK-ZK
Pacilatxel
Paciltaxal
Paciltaxel
Paciltaxle
Pacitaxel
Pacliatxel
Pacliltaxel
Paclitaxil
Palixtaxel
Poly LCLC
R04929097
RBBX 01
Ras Inhibitor
Reinoid/CIS retinoic acid
Rhumab/Ind
SAHA
SCH 58500
SCH63666
SCH6636
Sovatenib
Study drug AMG 655
TC
Tanceva
Taxoterecin
Temozolomoide
VP16
Vamydex
ac
anti neopastons
ch81c6
dcVax
letrozolum
metotreksat
mu81c6
paclitaxelum
rec MAGE 3-AS + AS15 ACS1 / Placebo Vaccine
rec MAGE3-AS+AS15 ASCI vs Placebo
recMAGE3-AS+AS15 ASCI/Placebo vaccine
recPRAME+AS15
recPRAME+AS15 ASCI
tamoxiphen+anastrazolum
tamoxiphene+anastrozolum
tc

File: transform/gdc/gdcutils.py [hunk 1]
```diff
@@ -46,3 +46,12 @@ def query_gdc(endpoint, params):
 
 def extract(data, keys):
     return {key: data.get(key) for key in keys}
+
+
+def get_file(file_id, path):
+    """ download a file from gdc, save in path """
```
Comment (adamstruck) [inline transform/gdc/gdcutils.py]: add a check so that it only downloads the file if a local copy doesn't exist

Comment (bwalsh) [reply transform/gdc/gdcutils.py]: done

**Pull request merged**